Framework/DataHandling/src/SinglePeriodLoadMuonStrategy.cpp

```diff
@@ -0,0 +1,185 @@
+
+  double firstGoodData =
+      LoadMuonNexusV2Helper::loadFirstGoodDataFromNexus(m_entry);
+  run.addProperty("FirstGoodData", firstGoodData);
+}
+
+/**
+ * Loads the good frames data into the stored workspace object
+ */
+void SinglePeriodLoadMuonStrategy::loadGoodFrames() {
+  // Overwrite existing log entry
+  auto &run = m_workspace->mutableRun();
+  run.removeProperty("goodfrm");
+
+  NXInt goodframes = LoadMuonNexusV2Helper::loadGoodFramesDataFromNexus(
+      m_entry, m_isFileMultiPeriod);
+
+  if (m_isFileMultiPeriod) {
+    run.addProperty("goodfrm", goodframes[static_cast<int>(m_entryNumber - 1)]);
+  } else {
+    run.addProperty("goodfrm", goodframes[0]);
+  }
+}
+
+/**
+ * Loads detector grouping.
+ * If no entry in NeXus file for grouping, load it from the IDF.
+ * stored
+ * @returns :: Grouping table
+ */
+Workspace_sptr SinglePeriodLoadMuonStrategy::loadDetectorGrouping() const {
+
+  auto grouping = LoadMuonNexusV2Helper::loadDetectorGroupingFromNexus(
+      m_entry, m_detectors, m_isFileMultiPeriod);
+  DataObjects::TableWorkspace_sptr table =
+      createDetectorGroupingTable(m_detectors, grouping);
+
+  Workspace_sptr table_workspace;
+  if (table->rowCount() != 0) {
+    table_workspace = std::dynamic_pointer_cast<Workspace>(table);
+  } else {
+    m_logger.notice("Loading grouping information from IDF");
+    table_workspace = loadDefaultDetectorGrouping();
+  }
+  return table_workspace;
+}
+/**
+ * Loads default detector grouping, if this isn't present
+ * return dummy grouping
+ * If no entry in NeXus file for grouping, load it from the IDF.
+ * @returns :: Grouping table
+ */
+Workspace_sptr
+SinglePeriodLoadMuonStrategy::loadDefaultDetectorGrouping() const {
+
+  auto instrument = m_workspace->getInstrument();
+  auto &run = m_workspace->mutableRun();
+  std::string mainFieldDirection =
+      run.getLogData("main_field_direction")->value();
+  API::GroupingLoader groupLoader(instrument, mainFieldDirection);
+  try {
+    const auto idfGrouping = groupLoader.getGroupingFromIDF();
+    return idfGrouping->toTable();
+  } catch (const std::runtime_error &) {
+    auto dummyGrouping = std::make_shared<Grouping>();
+    if (instrument->getNumberDetectors() != 0) {
+      dummyGrouping = groupLoader.getDummyGrouping();
+    } else {
+      // Make sure it uses the right number of detectors
+      std::ostringstream oss;
+      oss << "1-" << m_workspace->getNumberHistograms();
+      dummyGrouping->groups.emplace_back(oss.str());
+      dummyGrouping->groupNames.emplace_back("all");
+    }
+    return dummyGrouping->toTable();
+  }
+}
+/**
+ * Loads dead time table, if this isn't present
+ * @returns :: Dead time table
+ */
+Workspace_sptr SinglePeriodLoadMuonStrategy::loadDeadTimeTable() const {
+
+  auto deadTimes = LoadMuonNexusV2Helper::loadDeadTimesFromNexus(
+      m_entry, m_detectors, m_isFileMultiPeriod);
+  auto deadTimeTable = createDeadTimeTable(m_detectors, deadTimes);
```

Comment:
`return createDeadTimeTable(m_detectors, deadTimes)` should work here similar to above as `TableWorkspace` inherits from `Workspace`.